geode-core/src/distributedTest/java/org/apache/geode/internal/cache/PartitionedRegionClearDUnitTest.java

```diff
@@ -0,0 +1,171 @@
+    accessor = cluster.startServerVM(4, locatorPort);
+    client1 = cluster.startClientVM(5,
+        c -> c.withPoolSubscription(true).withLocatorConnection((locatorPort)));
+    client2 = cluster.startClientVM(6,
+        c -> c.withPoolSubscription(true).withLocatorConnection((locatorPort)));
+    dataStore1.invoke(this::initDataStore);
+    dataStore2.invoke(this::initDataStore);
+    dataStore3.invoke(this::initDataStore);
+    accessor.invoke(this::initAccessor);
+    client1.invoke(this::initClientCache);
+    client2.invoke(this::initClientCache);
+  }
+
+  protected RegionShortcut getRegionShortCut() {
+    return RegionShortcut.PARTITION_REDUNDANT;
+  }
+
+  private Region getRegion(boolean isClient) {
+    if (isClient) {
+      return getClientCache().getRegion(REGION_NAME);
+    } else {
+      return getCache().getRegion(REGION_NAME);
+    }
+  }
+
+  private void verifyRegionSize(boolean isClient, int expectedNum) {
+    assertThat(getRegion(isClient).size()).isEqualTo(expectedNum);
+  }
+
+  private void initClientCache() {
+    Region region = getClientCache().createClientRegionFactory(ClientRegionShortcut.CACHING_PROXY)
+        .create(REGION_NAME);
+    region.registerInterestForAllKeys(InterestResultPolicy.KEYS);
+  }
+
+  private void initDataStore() {
+    getCache().createRegionFactory(getRegionShortCut())
+        .setPartitionAttributes(new PartitionAttributesFactory().setTotalNumBuckets(10).create())
+        .addCacheListener(new CacheListenerAdapter() {
+          @Override
+          public void afterRegionDestroy(RegionEvent event) {
+            Region region = event.getRegion();
+            logger.info("Region " + region.getFullPath() + " is destroyed.");
+          }
+
+          @Override
+          public void afterRegionClear(RegionEvent event) {
+            Region region = event.getRegion();
+            logger.info("Region " + region.getFullPath() + " is cleared.");
+          }
+        })
+        .create(REGION_NAME);
+  }
+
+  private void initAccessor() {
+    getCache().createRegionFactory(getRegionShortCut())
+        .setPartitionAttributes(
+            new PartitionAttributesFactory().setTotalNumBuckets(10).setLocalMaxMemory(0).create())
+        .create(REGION_NAME);
+  }
+
+  private void feed(boolean isClient) {
+    Region region = getRegion(isClient);
+    IntStream.range(0, NUM_ENTRIES).forEach(i -> region.put(i, "value" + i));
+  }
+
+  @Test
+  public void normalClearFromDataStore() {
+    accessor.invoke(() -> feed(false));
+    dataStore1.invoke(() -> verifyRegionSize(false, NUM_ENTRIES));
+    dataStore2.invoke(() -> verifyRegionSize(false, NUM_ENTRIES));
+    dataStore3.invoke(() -> verifyRegionSize(false, NUM_ENTRIES));
+
+    dataStore1.invoke(() -> getRegion(false).clear());
+    dataStore1.invoke(() -> verifyRegionSize(false, 0));
+    dataStore2.invoke(() -> verifyRegionSize(false, 0));
+    dataStore3.invoke(() -> verifyRegionSize(false, 0));
```

Comment:
These verifications of region size on the data stores are called multiple times in multiple tests. Could they be extracted to helper methods to reduce code duplication?